src/Web/Slack/Pager/Types.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Web.Slack.Pager.Types where
+
+import Web.Slack.Prelude
+import Web.Slack.Util
+
+newtype Cursor = Cursor {unCursor :: Text}
+  deriving stock (Eq, Generic, Show)
+  deriving newtype (NFData, Hashable, FromJSON, ToJSON, ToHttpApiData)
+
+newtype ResponseMetadata = ResponseMetadata {responseMetadataNextCursor :: Maybe Cursor}
+  deriving stock (Eq, Show, Generic)
+
+instance NFData ResponseMetadata
+
+$(deriveJSON (jsonOpts "responseMetadata") ''ResponseMetadata)
```

Comment:
This is simple enough write the instances without the Template Haskell overhead? 

Or this is simple enough the Template Haskell overhead doesn't matter?

Reply:
The main motivation is that it was moved. I would rather take the TH overhead than do it inconsistently, I guess.